pkg/helm/actions/install_chart.go

```diff
@@ -1,22 +1,118 @@
 package actions
 
 import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/openshift/api/helm/v1beta1"
 	"github.com/openshift/console/pkg/helm/metrics"
 	"helm.sh/helm/v3/pkg/action"
 	"helm.sh/helm/v3/pkg/chart"
 	"helm.sh/helm/v3/pkg/chart/loader"
 	"helm.sh/helm/v3/pkg/release"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/klog"
+)
+
+var (
+	helmChartRepositoryClusterGVK = schema.GroupVersionResource{
+		Group:    "helm.openshift.io",
+		Version:  "v1beta1",
+		Resource: "helmchartrepositories",
+	}
+	helmChartRepositoryNamespaceGVK = schema.GroupVersionResource{
+		Group:    "helm.openshift.io",
+		Version:  "v1beta1",
+		Resource: "projecthelmchartrepositories",
+	}
 )
 
-func InstallChart(ns, name, url string, vals map[string]interface{}, conf *action.Configuration) (*release.Release, error) {
+func InstallChart(ns, name, url string, vals map[string]interface{}, conf *action.Configuration, client dynamic.Interface, coreClient corev1client.CoreV1Interface) (*release.Release, error) {
 	cmd := action.NewInstall(conf)
+	// tlsFiles contain references of files to be removed once the chart
+	// operation depending on those files is finished.
+	tlsFiles := []*os.File{}
+	var tlsConfigNamespace, configMapName, secretName string
+	repositoryName, _, err := getChartNameAndNamespaceFromChartUrl(url, ns, client, coreClient)
+	if err != nil {
+		//serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: fmt.Sprintf("Failed to parse request: %v", err)})
+		return nil, err
+	}
+	// Create a Kubernetes core/v1 client.
+	connectionConfig, err := getRepoConnectionConfig(repositoryName, ns, client)
+	if err != nil {
+		//serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: fmt.Sprintf("Failed to parse request: %v", err)})
+		return nil, err
+	}
+	if connectionConfig != (&v1beta1.ConnectionConfig{}) {
```

Comment:
`connectionConfig` is guaranteed to be a `*v1beta1.ConnectionConfig` because of `getRepoConnectionConfig`; this condition will never be true.